### PR TITLE
Add utility methods for Value/Put and a builder class for Key

### PR DESCRIPTION
**Pull request by brfrn169**
I added utility methods for Value/Put and a builder class for Key, which I believe will make Scalar DB easy to use.

### CODE REVIEW

File: src/main/java/com/scalar/db/api/Put.java [hunk 1]
```diff
@@ -51,6 +58,90 @@ public Put withValue(Value<?> value) {
     return this;
   }
 
+  /**
+   * Adds the specified BigInt value to the list of put values.
```
Comment (feeblefakie) [inline src/main/java/com/scalar/db/api/Put.java:62]: Just a minor question.
It seems these `withValue` methods are ordered by the types (?)
Is it common to do that?
(I just thought I would do with boolean -> int -> bigint -> float -> double -> text -> bytes or something, but I don't know that order is. )

**Review by feeblefakie — APPROVED** [review]
LGTM! Great improvement. Thank you.
Left one very minor question.

File: src/main/java/com/scalar/db/api/Put.java (hunk 1 above)
Comment (brfrn169) [reply src/main/java/com/scalar/db/api/Put.java:62]: Thank you for reviewing this!

Actually, I wasn't concerned about the method order very much. And it looks like the Google style plugin doesn't reorder the methods. Maybe I chose the alphabetical order (BigInt -> Blob -> Boolean -> Double -> Float -> Int -> Text). Should I reorder the methods?

And if we need to care about the method order, do we need to add this rule to our code rule? Or Google style already has this rule?

**Review by feeblefakie — APPROVED** [review]
LGTM! Thank you!

**Pull request merged**